src/services/messagesService.ts

```diff
@@ -229,18 +233,62 @@ const setConversationUnread = async function(token: string, options?: AxiosReque
  * @param fromMessageId The last read message to start from
  * @param [options] Axios request options
  */
-const summarizeChat = async function(token: string, fromMessageId: summarizeChatParams['fromMessageId'], options?: AxiosRequestConfig): summarizeChatResponse {
+async function summarizeChat(token: string, fromMessageId: summarizeChatParams['fromMessageId'], options?: AxiosRequestConfig): summarizeChatResponse {
 	return axios.post(generateOcsUrl('apps/spreed/api/v1/chat/{token}/summarize', { token }), {
 		fromMessageId,
 	} as summarizeChatParams, options)
 }
 
+/**
+ * Fetch a thread for given conversation and thread id
+ *
+ * @param token the conversation token
+ * @param threadId The thread id to retrieve data
+ * @param [options] Axios request options
+ */
+async function getSingleThreadForConversation(token: string, threadId: number, options?: AxiosRequestConfig): getThreadResponse {
+	return axios.get(generateOcsUrl('apps/spreed/api/v1/chat/{token}/threads/{threadId}', { token, threadId }), options)
+}
+
+/**
+ * Fetch a list of threads for given conversation
+ *
+ * @param data the wrapping object
+ * @param data.token the conversation token
+ * @param [data.limit=25] Number of threads to return
+ * @param [data.offsetId=0] The last thread ID that was known, default 0 starts from the newest
+ * @param [options] Axios request options
+ */
+async function getThreadsForConversation({ token, limit, offsetId }: { token: string } & listThreadsParams, options?: AxiosRequestConfig): listThreadsResponse {
+	return axios.get(generateOcsUrl('apps/spreed/api/v1/chat/{token}/threads', { token }), {
+		...options,
+		params: {
+			limit,
+			offsetId,
+		},
+	})
+}
+
+/**
+ * Create a new thread for a conversation
+ *
+ * @param token The conversation token
+ * @param messageId The message id of any message belonging to the future thread
+ * @param [options] Axios request options
+ */
+async function makeThreadForConversation(token: string, messageId: number, options?: AxiosRequestConfig): makeThreadResponse {
```

Comment:
Not sure you noticed, I chaged that in the last commits as well so it also returns TalkThreadInfo instead of TalkThread only